lib/swagger/paths.js

```diff
@@ -214,7 +231,7 @@ swaggerPaths._getPathOperationObject = function(options) {
     pathDefinition.responses["200"] = undefined;
   }
 
-  if ((options.handler !== "search") && (options.handler !== "create")) {
+  if (options.hasPathId) {
```

Comment:
When we create the path operation objects, we now pass in `hasPathId` which reflects if the URL contains a `/{id}/` parameter. We used to guess based on the operation, but it wasn't specific enough for all edge cases.